lib/str.js

```diff
@@ -935,6 +935,20 @@ const Str = {
         return CONST.SMS.E164_REGEX.test(phone);
     },
 
+    /**
+     * We validate mentions by checking if it's first character is valid for mention
+     * For example, *@username@expensify.com* is a valid mention because we allow bold styling for it.
+     * A valid mention starts with a space, *, _, #, ', ", or @ (means with no preceding characters).
```

Comment:
This is a great comment. Can we also add a line explaining that we don't support mentions within code or codefence?

Reply:
That works. Alternatively, you could do a more general comment for the function. Perhaps something like this: 

>We validate mentions by checking if it's first character is an allowed character and by checking that it isn't within other tags where mentions aren't allowed. 
>For example, *@username@expensify.com* is a valid mention because we allow bold styling for it, but `@username@expensify.com` is not because we do not allow mentions within code

Then above the `startsWithValidChar` variable you could have: 
>A valid mention starts with a space, *, _, #, ', ", or @ (with no preceding characters).

And above the `containsInvalidTag` variable you could have: 
>We don't support mention inside code or codefence styling, so using @username@expensify.com or @username@expensify.com will be invalid.

If we want to make it more explicit and clear what each is doing.